command/fs_ls.go

```diff
@@ -68,6 +74,20 @@ func (f *FSListCommand) Run(args []string) int {
 		return 1
 	}
 
+	// If -job is specified, use random allocation, otherwise use provided allocation
+	allocID := ""
+	if job {
+		jobID := args[0]
+		allocs, _, err := client.Jobs().Allocations(jobID, nil)
+		if err != nil {
+			f.Ui.Error(fmt.Sprintf("Error querying API: %v", err))
+		}
+		r := rand.New(rand.NewSource(time.Now().UnixNano()))
+		allocID = allocs[r.Intn(len(allocs))].ID
```

Comment:
What is odd is that this will potentially choose a defunct (stopped) allocation, when the task is actually running on a newer one. I think you'd want to look at the last evaluation, and only pick an allocation from its result. Correct me if I am wrong though, /cc @dadgar